engine/src/main/scala/cromwell/webservice/MetadataBuilderActor.scala

```diff
@@ -128,41 +128,60 @@ object MetadataBuilderActor {
       case s => Option(s.toInt)
     }
   }
+
+  private case object IndexedEmptyJson extends IndexedJsonValue {
+    override def index: Option[Int] = None
+    override def withIndex(index: String): IndexedJsonValue = this
+    override def toJson: JsValue = throw new IllegalStateException("IndexedEmptyJson cannot be converted to Json")
+  }
```

Comment:
Here's Wikipedia's menu of Group-like category theory [structures](https://en.wikipedia.org/wiki/Category_(mathematics)).  I thought this `IndexedEmptyJson` looked kind of like an identity element for `IndexedJsonValue`s.  An identity element is the only difference between the `Semigroup` currently used here and a `Monoid`.